patches/server/0970-New-specific-event-watching-command.patch

```diff
@@ -0,0 +1,220 @@
++                sender.sendMessage(text("Now watching event " + args[0] + "!", GREEN));
++            } catch(IOException ex) {
++                sender.sendMessage(text("Failed to create logging files! See the console for more info.", RED));
++                MinecraftServer.LOGGER.warn("Error occurred while creating event verbose files", ex);
++            }
++        }
++        return true;
++    }
++
++    /**
++     * A wrapper class for readability, essentially just handles storage of every Event we're watching for as well as
++     * that events verbose, and it's relevant file to store it
++     */
++    private static class Watcher {
++        private final String eventPath;
++        private final Path file;
++        private final List<Wrapper> wrappers;
++
++        private Watcher(String eventPath, Path file) {
++            this.eventPath = eventPath;
++            this.file = file;
++            this.wrappers = Lists.newArrayList();
++        }
++
++        public void log(Event ev, Listener l) {
++            wrappers.add(new Wrapper(ev, l));
++        }
++
++        public void dump() throws IOException{
++            JsonArray array = new JsonArray();
++            wrappers.forEach(w -> array.add(w.serialize()));
++
++            StringWriter stringWriter = new StringWriter();
++            JsonWriter jsonWriter = new JsonWriter(stringWriter);
++            jsonWriter.setIndent(" ");
++            jsonWriter.setLenient(false);
++            Streams.write(array, jsonWriter);
++
++            try (PrintStream out = new PrintStream(Files.newOutputStream(file), false, StandardCharsets.UTF_8)) {
++                out.print(stringWriter);
++            }
++        }
++    }
++
++    /**
++     * Wrapper class to store all relevant information
++     */
++    private static final class Wrapper {
++        private final Listener listener;
++        private final String timestamp;
++        private final Event event;
++
++        public Wrapper(Event event, Listener listener) {
++            this.listener = listener;
++            this.timestamp = FORMATTER.format(LocalDateTime.now());
++            this.event = event;
++        }
++
++        private JsonObject serialize() {
++            JsonObject parent = new JsonObject();
++            parent.addProperty("timestamp", timestamp);
++            parent.addProperty("listener", listener != null ? listener.getClass().getCanonicalName() : "INITIAL CALL");
++            JsonObject obj = new JsonObject();
++            obj.addProperty("event", event.getEventName());
++            try {
++                for (Field f : event.getClass().getDeclaredFields()) {
++                    if(f.getName().equalsIgnoreCase("handlers"))
++                        continue;
++                    f.setAccessible(true);
++                    Object field = f.get(event);
++                    obj.addProperty(f.getName(), field == null ? "null" : field.toString());
++                }
```

Comment:
Same thing here as for the other command. Since its almost identifal logic, it should be abstracted out. There's a lot of similarity between these events and therefore a ton of duplicated logic.